n3fit/src/n3fit/model_gen.py

```diff
@@ -671,153 +773,119 @@ def compute_unnormalized_pdf(x):
                 "photon_integral": photon_integrals,
             }
         )
-        PDFs = PDFs_normalized
     else:
         PDFs = PDFs_unnormalized
+        sumrule_layer = lambda x: x
 
+    ## Include the photon in the PDF for QED-enabled fits
+    # (by default the entry corresponding to the photon is set to 0)
     if photons:
+        layer_photon = AddPhoton(photons=photons, name="add_photon")
         PDFs = layer_photon(PDFs)
 
+    # Return a PDF without a replica axis, to extract single replicas from an ensemble
     if not replica_axis:
         PDFs = Lambda(lambda pdfs: pdfs[:, 0], name="remove_replica_axis")(PDFs)
 
-    pdf_model = MetaModel(model_input, PDFs, name=f"PDFs", scaler=scaler)
-    return pdf_model
+    return MetaModel(model_input, PDFs, name=f"PDFs", scaler=scaler)
 
 
-def generate_nn(
-    layer_type: str,
-    nodes_in: int,
-    nodes: list[int],
-    activations: list[str],
-    initializer_name: str,
-    replica_seeds: list[int],
-    dropout: float,
-    regularizer: str,
-    regularizer_args: dict,
-    last_layer_nodes: int,
+# TODO: is there a way of keeping sincronized the input of this function and ReplicaSettings
+# beyond a test of it? In principle we might want to have the arguments explicitly here...
+def _generate_nn(
```

Comment:
Why doesn't this simply take `ReplicaSetting` as part of the argument? This would simplify things and prevents doing something like this for example:
```Python
_generate_nn(x_input, i, **asdict(replica_settings))
```